zilliqa/src/contracts/deposit.sol

```diff
@@ -160,6 +166,9 @@ contract Deposit {
 
         _stakersMap[blsPubKey].rewardAddress = rewardAddress;
         _stakersMap[blsPubKey].peerId = peerId;
+        _stakersMap[blsPubKey].signerAddress = signerAddress;
+
+        emit StakerAdded(blsPubKey);
     }
 
     function setStake(
```

Comment:
This is be calculated when using z2 so it doesn't need to be specified independently.

Reply:
Sorry I don't understand what you mean? Either way I still think a clarifying comment here is important if `_stakersMap[blsPubKey].signerAddress = address(0)` (i.e. leaving it at the default) is intended. Otherwise, it easily looks like we've just forgotten to set a field that is set in `deposit`.

Reply:
Please don't resolve this conversation without discussing or making any changes. Feel free to disagree with me.

Reply:
I'll resolve this based on what @DrZoltanFazekas responds about whether it's optional or not.

Reply:
Also validators added by `setStake()` need to have a `signerAddress` to participate in UCCB, which is mandatory for all validators

Reply:
Added an assert that signerAddress != address(0).

Reply:
I don't think that's what we need. There's no inherent reason we should disallow a `signerAddress` of zero (except that its probably useless, but that's the user's fault). Instead, we need to make sure we *do* assign the `signerAddress` in `setStake`, so that it doesn't remain default-initialized to zero.

Reply:
The change to setStake was already done in a subsequent PR. I've moved the change here.
I'm still leaving the check for `signerAddress != address(0)` because it seems like the correct thing to do. Leaving it as 0 probably wouldn't be intended and I prefer to avoid it (or any other issues that might arise from accidentally forgetting to set it).

Reply:
The change to setStake was already done in a subsequent PR; I've moved it here.